package AIT.protocols.csp.abt;

import "common-protocols.proto";

message P_EndPoint {
  required int32 id = 1;
  required string host = 2;
  required int32 port = 3;
}

enum CP_MessageType {
  T_INTRODUCE = 1;
  T_INTRODUCE_ACK = 2;
  T_REQUEST_LIST = 3;
  T_REQUEST_ACK = 4;
  T_LIST = 5;
  ERR_NO_MORE_AGENTS = 6;
  ERR_REPEATED_ID = 7;
  ERR_UNKNOWN = 8;
}

message P_CommunicationProtocol {
  required CP_MessageType type = 1;
  optional int32 id = 2;
  optional P_EndPoint identity = 3;
  repeated P_EndPoint others = 4;
}

enum P_MessageType {
  T_OK = 1;
  T_NOGOOD = 2;
  T_ADDLINK = 3;
  T_STOP = 4;
}

message P_Message {
    required P_MessageType type = 1;
    required int32 sender = 2;
    optional P_Assignment assignment = 3;
    optional P_Nogood nogood = 4;
}

message P_Nogood {
    required P_CompoundAssignment lhs = 1;
    required P_Assignment rhs = 2;
}
